lib/app.js

```diff
@@ -34,22 +35,53 @@ module.exports = function (options) {
       req.url = path.join('/', host, req.url);
     }
 
-    if (options.cors) {
-      if (req.method === 'OPTIONS') {
-        if (req.headers['access-control-request-headers'])
-          res.header('Access-Control-Allow-Headers', req.headers['access-control-request-headers']);
-        if (req.headers['access-control-request-method'])
-          res.header('Access-Control-Allow-Methods', req.headers['access-control-request-method']);
+    next();
+  });
+
+  let CORSConfiguration;
+  if (typeof options.cors === 'string') {
+    const corsConfigFile = fs.readFileSync(options.cors);
+    xml2js.parseString(corsConfigFile, { async: false }, (err, parsed) => {
+      if (err) {
+        err.name = 'Invalid CORS XML';
+        throw err;
       }
-      if (req.headers.origin)
-        res.header('Access-Control-Allow-Origin', '*');
-    }
+      ({ CORSConfiguration } = parsed);
+    });
+  } else {
+    ({ CORSConfiguration } = options.cors);
```

Comment:
Since you mention it, we should probably have at least a couple tests for using the actual CLI, seeing that's probably how most people use this project.

Anyway I think I forgot to test that, I think you're right in that it will be set to `false` which can cause problems. I will have to make some changes there. I also found the way I have the default config specified kind of awkward since it can be either an object or a string. Do you think it would be better to use the path to the sample config xml as a default value instead?

Reply:
I agree about testing the actual CLI, I don't really have any ideas on how to write those tests right now though. Definitely need to check for `false` and `null` as opposed to all falsey values because I'd assume you'd want to set the default when the user merely leaves `options.cors` undefined.

I think using object by default is fine, users would need to do more searching to find the default config XML file rather than seeing it directly in the source code. However it would be great to see the default CORS configuration XML included or linked in the documentation so people can quickly copy and paste it into their own projects if they want to.